src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.VirtualMonoRepo.E2E.Tests/VmrSyncRepoChangesE2ETest.cs

```diff
@@ -0,0 +1,211 @@
+
+        var expectedFilesFromRepos = new List<LocalPath>
+        {
+            testRepoFilePath,
+            dependencyFilePath
+        };
+
+        var expectedFiles = GetExpectedFilesInVmr(
+            _vmrPath,
+            new[] { "test-repo", "dependency" },
+            expectedFilesFromRepos
+        );
+
+        CheckDirectoryContents(_vmrPath, expectedFiles);
+        CheckFileContents(testRepoFilePath, "Test changes in repo file");
+        CheckFileContents(dependencyFilePath, "File in dependency");
+        await CheckAllIsCommited(_vmrPath);
+    }
+
+    [Test]
+    public async Task FileIsIncludedTest()
+    {
+        Directory.CreateDirectory(_privateRepoPath / "excluded");
+        File.WriteAllText(_privateRepoPath / "excluded" / "excluded.txt", "File to be excluded");
+        await CommitAll(_privateRepoPath, "Create an excluded file");
+
+        await EnsureTestRepoIsInitialized();
+
+        File.Move(_privateRepoPath / "excluded" / "excluded.txt", _privateRepoPath / "excluded.txt");
+        await CommitAll(_privateRepoPath, "Move a file from excluded to included folder");
+
+        await UpdateRepoToLastCommit("test-repo", _privateRepoPath);
+
+        var expectedFilesFromRepos = new List<LocalPath>
+        {
+            _vmrPath / "src" / "test-repo" / "test-repo-file.txt",
+            _vmrPath / "src" / "dependency" / "dependency-file.txt",
+            _vmrPath / "src" / "test-repo" / "excluded.txt",
+        };
+
+        var expectedFiles = GetExpectedFilesInVmr(
+            _vmrPath,
+            new[] { "test-repo", "dependency" },
+            expectedFilesFromRepos
+        );
+
+        CheckDirectoryContents(_vmrPath, expectedFiles);
+        CheckFileContents(_vmrPath / "src" / "test-repo" / "excluded.txt", "File to be excluded");
+        await CheckAllIsCommited(_vmrPath);
+    }
+
+    [Test]
+    public async Task SubmodulesAreInlinedProperlyTest()
+    {
+        var testRepoFilePath = _vmrPath / "src" / "test-repo" / "test-repo-file.txt";
+        var dependencyFilePath = _vmrPath / "src" / "dependency" / "dependency-file.txt";
+        var submoduleFilePath = _vmrPath / "src" / "test-repo" / "externals" / "external-repo" / "external-repo-file.txt";
+        var additionalSubmoduleFilePath = _vmrPath / "src" / "test-repo" / "externals" / "external-repo" / "additional-file.txt";
+
+        await EnsureTestRepoIsInitialized();
+
+        var submoduleRelativePath = new NativePath("externals") / "external-repo";
+        await InitializeSubmodule(_privateRepoPath, "submodule1", _externalRepoPath, submoduleRelativePath);
+        await CommitAll(_privateRepoPath, "Add submodule");
+        await UpdateRepoToLastCommit("test-repo", _privateRepoPath);
+
+        var expectedFilesFromRepos = new List<LocalPath>
+        {
+            testRepoFilePath,
+            dependencyFilePath,
+            submoduleFilePath,
+            _vmrPath / "src" / "test-repo" / ".gitmodules",
```

Comment:
I wouldn't test for this file. We don't care about it and in future we might even remove these and we don't care
```suggestion
```